### DISCUSSION TITLE
How to mount folder using ruri? / 如何挂载目录？

### DISCUSSION THREAD

**Discussion by Young-Lord** [RuriOSS/daijin#46] [Q&A] [answered]
我在 Termux 环境中使用``su -c `which ruri` -p -m /data/data /data/data /data/kali``试图挂载`/data/data`，但貌似并没有用
git commit id: `c7c6d631aa12072c2acbd35f232d12a546de02f7`
```
Script Info :
[] Author                : Moe-hacker
[] License               : APACHE-2.0
[] Version               : 9.0-dev
[] ruri Version :（这里没输出，根据`ruri -v`输出是`2.0-dev`）
[] ruri License : MIT

System info :
[] Kernel Version        : 4.19.157
[] Termux Version        : 0.118.0
[] Username              : [REDACTED]
[] Android Version       : 13
[] Cpu Architecture      : aarch64
[] Hostname              : localhost
[] Uptime                : [REDACTED]
[] Magisk version        : 26103
[] SELinux               : Enforcing

Hardware info :
[] Brand                 : Xiaomi
```

**Comment by Young-Lord** [comment]
权限等级 2，`chroot`容器，未开启`unshare`
以及，目标是调用`termux-api`里的脚本

**Comment by Young-Lord** [comment]
在 ruri `a6a1570`（此时的最新`commit`），问题依然存在

**Comment by Moe-hacker** [comment]
C语言的mkdir()不支持创建多级目录，测试一直用的/mnt->/mnt所以没发现，手动创建下挂载点应该就能跑了，在修了。。。

**Reply by Young-Lord** [reply]
> C语言的mkdir()不支持创建多级目录，测试一直用的/mnt->/mnt所以没发现，手动创建下挂载点应该就能跑了，在修了。。。

已经手动创建过`/data/data`，貌似还是不行（表现为空目录）

**Reply by Moe-hacker** [reply, marked as answer]
最新的提交(f64abd6)应该已经修复了，至少在我手机和电脑上测试是这样，同时感谢反馈，一个人测试难免有疏漏😂
~~话说一个20行的mkdir -p我居然写了一个小时（指针用惯了不会用数组操作字符了）。。。。。~~

**Reply by Young-Lord** [reply]
> 最新的提交(f64abd6)应该已经修复了，至少在我手机和电脑上测试是这样，同时感谢反馈，一个人测试难免有疏漏😂
> ~~话说一个20行的mkdir -p我居然写了一个小时（指针用惯了不会用数组操作字符了）。。。。。~~

好，它工作！
以及，我在`termux-container`的	`submodule`里更新`ruri`到现在`main`分支上的版本后，貌似`make install`就炸了（`ld.lld: error: unable to find library -lpthread`，并且没有继续安装其它文件）

**Reply by Moe-hacker** [reply]
termux-container目前还不太稳定,后续会重写一遍 ~~(现在这一版能跑已经是奇迹了)~~
